### PR TITLE
Updates config to use orb

### PR DESCRIPTION
**Pull request by kierangillen**
- Updates `config.yml` to use hokusai orb

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,53 +1,53 @@
...
+version: 2.1
+
+orbs:
+  hokusai: artsy/hokusai@0.3.0
+
+not_staging_or_release: &not_staging_or_release
+  filters:
+    branches:
+      ignore:
+        - staging
```
Comment (ashkan18) [inline .circleci/config.yml]: we don't have `staging` or `release`. Looking at this, we may want to create them now.

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,53 +1,53 @@
...
+          project-name: exchange
           requires:
-            - push
+            - push-staging-image
+
+      # release
+      - hokusai/push:
+          name: push-production-image
+          <<: *only_release
+      - hokusai/deploy-production:
```
Comment (ashkan18) [inline .circleci/config.yml]: nice! currently we deploy using `hokusai pipeilne promote` if the plan is to switch to PR-based deploy as part of this, we should update readme https://github.com/artsy/exchange/blob/master/README.md#production

Comment (kierangillen) [reply .circleci/config.yml]: Just did that, thanks!

File: .circleci/config.yml (hunk 1 above)
Comment (kierangillen) [reply .circleci/config.yml]: We think that github will create them automatically. We will verify.

**Pull request merged**